### PR TITLE
Anthony McKeever - Lesson 3 Exercises 1 through 3

### PR DESCRIPTION
**Pull request by Snip3rM00n**
All the exercises for Lesson 3.  This does not include the assignment, that will be in the next pull request.

I found Task 2, the string format lab, to be a little challenging.  I feel I did an okay job, however I am open to feedback on it.  :)

**_Squashed Commit History:_**

* Adds Slice Lab

A lab about slicing sequences (not tasty beverages)
Adds unit tests.
Updates readme

* Save changes

* lesson 3 start

* Finish Exercise 2 Tasks 1 - 6

Formatting Strings
Formatting integers into strings
Rearanging integers
Creating tables
Weighing fruit
	100% more vitamin C

* Add List Lab

Printing Fruit
Deleting Fruit
Getting hungry
Reversing Fruit
Asking about Fruit

* Implement self feedback

### CODE REVIEW

File: students/anthony_mckeever/lesson3/exercise_1/slice_lab.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    print("\tTesting Lists")
+    assert reorder_sequence([0, 1, 2, 3, 4, 5, 6, 7, 8]) == [6, 7, 8, 0, 1, 2, 3, 4, 5]
+    assert reorder_sequence([1, 2, 3]) == [3, 1, 2]
+    assert reorder_sequence([1, 2]) == [1, 2]
+
+    print("\tTesting Tuples")
+    assert reorder_sequence(("Pikachu", "Mareep", "Voltorb", "Rotom", "MissingNo.", "Pichu")) == ( "MissingNo.", "Pichu", "Pikachu", "Mareep", "Voltorb", "Rotom")
+    assert reorder_sequence(("Mareep", "Flaafy", "Ampharos")) == ("Ampharos", "Mareep", "Flaafy")
+    assert reorder_sequence((True, False)) == (True, False)
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson3/exercise_1/slice_lab.py]: Looks good - nice assert statements!

File: students/anthony_mckeever/lesson3/exercise_2/strformat_lab.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    input_table, __ = get_table_test_case()
+    print("\nTask 6:")
+    print(create_table(input_table))
+
+if __name__ == "__main__":
+    print("Running Tests...")
+    run_tests()
+
+    print("\nPrinting Exercises...")
+    print_tasks()
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson3/exercise_2/strformat_lab.py]: Looks great!

File: students/anthony_mckeever/lesson3/exercise_3/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+print("Series 1: Fun with Fruit!\n")
+
+fruits_list = ["Apples", "Pears", "Oranges", "Peaches"]
+print_fruit(fruits_list)
+
+fruits_list.append(get_fruit())
+print_fruit(fruits_list)
+
+fruit = input(f"Choose a number between 1 and {len(fruits_list)} > ")
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson3/exercise_3/list_lab.py]: Nice use of len()

File: students/anthony_mckeever/lesson3/exercise_3/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+print(f"You chose {fruit}: {fruits_list[int(fruit) - 1]}")
+
+fruits_list = [get_fruit()] + fruits_list
+print_fruit(fruits_list)
+
+fruits_list.insert(0, get_fruit())
+print_fruit(fruits_list)
+
+for fruit in fruits_list:
+    if fruit[0] == "P":
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson3/exercise_3/list_lab.py]: I'd use lower() or upper() to catch case sensitive user input

File: students/anthony_mckeever/lesson3/exercise_3/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+for fruit in fruits_list:
+    choice = input(f"Do you like {fruit}? > ")
+    
+    if choice.lower() not in ["yes", "no"]:
+        while choice.lower() not in ["yes", "no"]:
+            choice = input("A simple \"yes\" or \"no\" please... > ")
+    elif choice.lower() == "yes":
+        fruits_list3.remove(fruit)
+else:
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson3/exercise_3/list_lab.py]: Hmm... this should be nested under your for loop, but it still works.  Interesting.  Recommend refactoring.

File: students/anthony_mckeever/lesson3/exercise_3/list_lab.py (hunk 4 above)
Comment (Snip3rM00n) [reply students/anthony_mckeever/lesson3/exercise_3/list_lab.py]: That is a great point!  I'll be sure to catch that in the future. :)

File: students/anthony_mckeever/lesson3/exercise_3/list_lab.py (hunk 5 above)
Comment (Snip3rM00n) [reply students/anthony_mckeever/lesson3/exercise_3/list_lab.py]: I did actually mean this else to be the end of the For Loop to ensure it was entered when the for loop naturally came to an end.  However I can also see now that that would have been unnecessary given that there is no `break` in the loop.

Looks like I slightly overthought that. :)

**Comment by Snip3rM00n** [comment]
@charlesrobison - My apologies.  It looks like when I merged my assignment branch into my personal master that it appended my assignment to this PR.  How would you like to proceed?

File: students/anthony_mckeever/lesson3/exercise_3/list_lab.py (hunk 5 above)
Comment (Snip3rM00n) [reply students/anthony_mckeever/lesson3/exercise_3/list_lab.py]: Refactored. :)

**Pull request merged**